app/_components/PoiCard.tsx

```diff
@@ -86,6 +88,7 @@ export function PoiCard({
         }
       );
       const JSONresponse = response.json() as Promise<Response>;
+      setCheckLevel(true); //to fetch level after guess
       return JSONresponse;
     } catch (error) {
       console.error(error);
```

Comment:
should use `finally` here since if there is some error occur in `try` block, the setChecklevel will still switch to true and never switch back to false.

```suggestion
      return JSONresponse;
    } catch (error) {
      console.error(error);
    } finally {
       setCheckLevel(true); //to fetch level after guess
    }
```